### PR TITLE
add tests on sets

### LINKED ISSUE #1359: [FEATURE] Introduce comprehensive tests for the set operators
To improve the coverage

### PR DESCRIPTION
**Pull request by konnov**
Closes #1359. A large test suite that is testing all set operators, especially in the corner cases. This test suite surfaced a string of bugs: #1361, #1360, #1358. Most of them are related to infinite sets and powersets.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: test/tla/TestSets.tla [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    /\ TestChooseEmpty
+    /\ TestChooseSet
+    /\ TestIsFiniteSet
+    \* IGNORE UNTIL FIXED: /\ TestIsFiniteSetOnInfinite
+    \* IGNORE UNTIL FIXED: /\ TestPowersetCardinality
+    \* IGNORE UNTIL FIXED: /\ TestPowersetEmpty
+    \* IGNORE UNTIL FIXED: /\ TestPowerset26
+    \* IGNORE UNTIL FIXED: /\ TestPowersetSubsetEq
+    \* IGNORE UNTIL FIXED: /\ TestExistsEmptyInPowerset
+    \* IGNORE UNTIL FIXED: /\ TestExists15InPowerset
```
Comment (thpani) [inline test/tla/TestSets.tla:250]: Is there a way to make sure we do not to miss uncommenting these once the issues are fixed?

File: test/tla/TestSets.tla [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+TestExistsSet ==
+    \E S \in { SetEmpty, Set26, Set1357 }:
+        5 \in S
+
+TestIsFiniteSet ==
+    /\ IsFiniteSet(SetEmpty)
+    /\ IsFiniteSet(Set26)
+    /\ IsFiniteSet(Set1357)
+
+TestIsFiniteSetOnInfinite ==
```
Comment (thpani) [inline test/tla/TestSets.tla:208]: This is also commented out below. Can you file and issue and/or link it?

**Review by thpani — APPROVED** [review]
2 small comments. Otherwise, LGTM

File: test/tla/TestSets.tla (hunk 1 above)
Comment (konnov) [reply test/tla/TestSets.tla:250]: There are issues for all of these tests, so we will uncomment them once the issues are fixed

File: test/tla/TestSets.tla (hunk 2 above)
Comment (konnov) [reply test/tla/TestSets.tla:210]: Good point! I've forgotten to add the link. Fixed

**Review by gabrielamafra — APPROVED** [review]
